forge-game/src/main/java/forge/game/ability/effects/AnimateEffect.java

```diff
@@ -40,6 +40,10 @@ public void resolve(final SpellAbility sa) {
             return;
         }
 
+        if ("UntilUntaps".equals(sa.getParam("Duration")) && !source.isTapped()) {
+            return;
+        }
```

Comment:
would it make sense to move the special "Until" Duration checks into SpellAbilityEffect base class?